### DISCUSSION TITLE
App Directory - Prevent routes from being added on first load for some route

### DISCUSSION THREAD

**Discussion by kavitang** [vercel/next.js#51623] [Help]
### Summary

We have the following structure in our app

```
/app
  (group1)
       /admin
          - layout.tsx
          - page.tsx
       /path1
          - layout.tsx
          - page.tsx
       - layout.tsx  
  (group2)
       /[[...slug]]
          - layout.tsx
          - page.tsx
       /path4
          - layout.tsx
          - page.tsx
       - layout.tsx  
  /path3
  layout.tsx (root)
```
All paths within `(group2)` are statically built and served. Currently when we load `myapp.com/path4` Next is adding some paths from `(group1)` even though there are no direct imports, Links, redirects to these routes. This is resulting in a lot of unnecessary JS being downloaded for `myapp.com/path4`, which is not ideal for mobile users as `(group1)` in our case has some very heavy code/functionality involving heavy packages.

How can we identify why Next is appending some routes from `(group1)` when loading (not navigating to) `path4` directly. It would be great if someone can explain how Next decides to append routes when loading a specific routes, beyond prefetching Links and wrapper layouts. Is there any specific file in the `.next` folder we could check to understand the tree from `/path4`?

Also since `(group2)/[[...slug]]` is at the root level, we would like to redirect users coming from a different domain (also serving the same app) to a different path. If this redirect, say to `/admin`, was added in `middleware.ts`  would it still be included when loading `/path4`?

The goal is to only load `(group2)` and related layouts (plus root layout) when any path under it is accessed. How would this be achieved? 

Edit: Tried using multiple root layouts

```
/app
  (group1)
       /admin
          - layout.tsx
          - page.tsx
       /path1
          - layout.tsx
          - page.tsx
       - layout.tsx  
      /path3
     - layout.tsx (Layout for group1)
  (group2)
       /[[...slug]]
          - layout.tsx
          - page.tsx
       /path4
          - layout.tsx
          - page.tsx
       - layout.tsx  (Layout for group2)
```
Even now when `/path4` is directly loaded, next downloads the paths from `(group1)` as well. In the JS downloaded for `/path4` I can see packages used in `(group1)/layout.tsx`. 

Would really appreciate some clarity on how these routes are added, or links to sources I can use to understand this better.

Thanks!

### Additional information

```js
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
    Binaries:
      Node: 18.14.2
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.4.5
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.8.4
```

**Comment by nkeat12** [comment]
Would love to see an answer on this @leerob or anyone else on the team!
